imports/models/BoardFavoritesCollection.ts

```diff
@@ -0,0 +1,50 @@
+import { BaseCollection } from './BaseCollection'
+import { BoardRecord } from './BoardCollection'
+import { Meteor } from 'meteor/meteor'
+
+export interface BoardFavoriteRecord {
+  userid: string
+  boardId: string
+}
+
+export class BoardFavoritesCollectionClass extends BaseCollection<BoardFavoriteRecord> {
+  constructor() {
+    super('board_favorites')
+  }
+
+  public async addFavoriteBoard(userid: string, boardId: string): Promise<boolean> {
+    const query = { userid, boardId }
+    const update = { $set: { userid, boardId } }
+    const result = await this.collection.upsertAsync(query, update)
+    if (result && result.numberAffected && result.numberAffected > 0) {
+      return result?.numberAffected > 0
+    } else {
+      throw new Meteor.Error(`Failed to add favorite board for user ${userid} and board ${boardId}`)
+    }
+  }
```

Comment:
```typescript
  public async addFavoriteBoard(userid: string, boardId: string): Promise<boolean> {
    const query = { userid, boardId }
    const update = { $set: { userid, boardId } }
    const result = await this.collection.upsertAsync(query, update)

    return result?.numberAffected > 0
  }
```

use this way

Reply:
![image](https://user-images.githubusercontent.com/92238941/230037240-48e07d83-6687-4a10-ab1b-4553395908cb.png)

Reply:
You can add a default value for the numberAffected, an example way:

```typescript
...
const numberAffected = result?.numberAffected ?? 0
return numberAffected > 0
```

Reply:
no it's working now